contrib_bots/lib/CommuteBot/CommuteBot.py

```diff
@@ -0,0 +1,215 @@
+                # ensures that invalid inputs are not entered into url request
+                if len(org) != 2:
+                    continue
+                key, value = org
+                params.update({key: value})
+
+            # limited to UTC timezone because of difficulty with user inputting
+            # correct string for input
+            if 'departure_time' in params:
+                if params['departure_time'] != 'now':
+                    departure_time = params['departure_time']
+                    times = departure_time.split(',')
+                    times = [int(x) for x in times]
+                    a = dt.datetime(1970, 1, 1, 0, 0, 0)
+                    b = dt.datetime(*times)
+                    total_seconds = str(int((b-a).total_seconds()))
+                    params.update({'departure_time': total_seconds})
+
+            if 'arrival_time' in params:
+                arrival = params['arrival_time']
+                times = arrival.split(',')
+                times = [int(x) for x in times]
+                a = dt.datetime(1970, 1, 1, 0, 0, 0)
+                b = dt.datetime(*times)
+                total_seconds = str(int((b-a).total_seconds()))
+                params.update({'arrival_time': total_seconds})
+
+            # adds API Authentication Key to url request
+            config = cp.ConfigParser()
+            home = expanduser('~')
+            settings = open(home + '/settings.ini')
+            config.readfp(settings)
+            api_key = config.get('Google.com', 'api_key')
+            params.update({'key': api_key})
+            settings.close()
+
+            # sends url api request and converts response into json format
+            r = requests.get('https://maps.googleapis.com/maps/api/'
+                             + 'distancematrix/json',
+                             params=params)
+            if r.status_code == 200:
+                rjson = r.json()
+            else:
+                self.send_info(message,
+                               "Something went wrong. Please try again."
+                               + " Error: {error_num}."
+                               .format(error_num=r.status_code), client)
+                return None
+
+            # determines if user has valid inputs
+            try:
+                not_found = (rjson["rows"][0]["elements"][0]["status"]
+                             == "NOT_FOUND")
+                invalid_request = (rjson["status"] == "INVALID_REQUEST")
+                no_result = (rjson["rows"][0]["elements"][0]["status"]
+                             == "ZERO_RESULTS")
+
+                if no_result:
+                    self.send_info(message,
+                                   "Zero results\n"
+                                   + "If stuck, try '@commute help'.", client)
+                    return None
+
+                elif not_found or invalid_request:
+                    raise Exception()
```

Comment:
You can raise `IndexError()` since that will allow you to move to the `except` part